deepfence_ui/app/scripts/components/integration-view/aws-s3-integration-view/add-form.js

```diff
@@ -1,124 +1,155 @@
+/* eslint-disable no-unused-vars */
 /* eslint-disable react/destructuring-assignment */
-import React from 'react';
+import React, { useState } from 'react';
 import withIntegrationForm from '../../../hoc/notification-integration';
 
-class AWSS3IntegrationForm extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.handleChange = this.handleChange.bind(this);
+const AWSS3IntegrationForm = props => {
+  const [state, setState] = useState({
+    bucketName: '',
+    s3FolderPath: '',
+    awsAccessKey: '',
+    awsSecretKey: '',
+    awsRegion: '',
+    integration_type: 's3',
+  });
 
-    this.state = {
-      bucketName: '',
-      s3FolderPath: '',
-      awsAccessKey: '',
-      awsSecretKey: '',
-      awsRegion: '',
-      integration_type: 's3',
-    };
-  }
-
-  handleChange(e) {
+  const handleChange = e => {
     const { name, value } = e.target;
-    this.setState({ [name]: value }, () => {
-      const state = {
-        ...this.state,
-        s3_bucket: this.state.bucketName,
-        folder_path: this.state.s3FolderPath,
-        aws_access_key: this.state.awsAccessKey,
-        aws_secret_key: this.state.awsSecretKey,
-        region_name: this.state.awsRegion,
-      };
-      this.props.saveChildFormData(state);
-    });
-  }
+    setState(prevState => ({
+      ...prevState,
+      [name]: value,
+    }));
+    props.saveChildFormData(state);
```

Comment:
it is possible that the state is not updated at this point, lets use second callback parameter on setState to do this.